### ISSUE TITLE
Resource files are not getting deleted from the LOD after the lesson has been set to not being visible

### ISSUE THREAD

**Issue by pcenov** [learningequality/kolibri#11558]
## Observed behavior

The resources from a lesson which has been changed back to not being visible to learners are not completely removed from the device and keep syncing back to the LOD after having been manually deleted from the device.

https://github.com/learningequality/kolibri/assets/79847249/e2b0308e-a691-4a24-9e09-e2471dd0f8fc

## Expected behavior

Any resources from a lesson not visible to the learners should be removed from the LOD and should not be auto-downloaded to the device once having been manually deleted.

## Steps to reproduce the issue
1. Install the [latest Beta 9 build asset](https://github.com/learningequality/kolibri/releases/tag/v0.16.0-beta9).
2. Setup a server and a LOD.
3. Assign a lesson to the LOD and wait for it to sync.
4. As a coach make the lesson not visible and confirm that after a successful sync it disappears from the LOD's home page. Observe that the channel resources remain on the device.
5. On the LOD go to Device > Channels and delete the channel as it's no longer needed on the device.
6. Observe that the channel gets synced back to the device.

## Logs

[UbuntuServer.zip](https://github.com/learningequality/kolibri/files/13460571/UbuntuServer.zip)
[UbuntuLOD.zip](https://github.com/learningequality/kolibri/files/13460573/UbuntuLOD.zip)

## Usage Details
Ubuntu 22  - Chrome

**Comment by pcenov** [comment]
@radinamatic

**Comment by Baizkunthaz** [comment]
Hey @pcenov, I am very willing and interested in contributing to the org, but as I am new to open source, could you please guide me on how to start?

**Comment by marcellamaki** [comment]
Hi @Baizkunthaz - thank you for your interest in contributing to Kolibri! We have a growing group of active, open source contributors here and we are very happy to have you. You can get started by reading the [Kolibri Dev Documentation](https://kolibri-dev.readthedocs.io/en/develop/index.html), which has important details about contributing, getting your local environment set up, and information about the Kolibri architecture. Please also review our [Github Discussions page for Kolibri](https://github.com/learningequality/kolibri/discussions), particularly [the welcome message which has information about getting started with open source contributing](https://github.com/learningequality/kolibri/discussions/10691). If you have questions about contributing, please ask them there. 

To find issues to get started with, look for [the "help wanted" label on open issues](https://github.com/learningequality/kolibri/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22), where no person is the "assignee".

**Comment by marcellamaki** [comment]
@pcenov - this is actually the expected behavior, I believe (cc @bjester to confirm). The lesson itself should no longer be visible with the class page, but the resources themselves are not deleted until there is a space issue.

**Comment by pcenov** [comment]
Hi @marcellamaki - notice that the text in the 'Make lesson not visible' modal states: 'Resource files in this lesson will be removed from learn-only devices that are set up to sync with this server.' Also the problem is that even after I manually delete the resources from the LOD device they get auto-synced back to it while the lesson is still marked as not visible. In my case I had a lesson with 1GB of resources so it auto-downloaded it to all of my LOD devices several times before I was able to identify this issue...

**Comment by marcellamaki** [comment]
Thanks @pcenov - that is very helpful detail. I'll chat with Blaine about this and come back to you with any more questions!

**Comment by bjester** [comment]
> this is actually the expected behavior, I believe (cc @bjester to confirm)

yes this is expected behavior: the resources are not deleted on the LOD until the space is needed for new assignments

although, as @pcenov pointed out, it shouldn't re-download the resources when the lesson is marked as inactive.

**Comment by marcellamaki** [comment]
I do think, reviewing @pcenov's comment, that there is a mis-match between the "expected behavior" that is explained to the user (i.e. resources will be removed), vs. the "expected behavior" in terms of how the code was structured to work, meaning no removal until space is needed. I'm not sure if there are some ways we could adjust those strings in the near term, I will look into it, as that might help with at least some of the confusion on the user end. In addition to the "re-dowloading" problem.

**Comment by bjester** [comment]
@marcellamaki Well the copy shown to the coach isn't necessarily wrong because it doesn't say 'when' the deletion happens, but yes it's easy for them to assume the 'when'. I think it aligns with our goals for Kolibri as offline software, because bandwidth comes at a cost, and [it's how the architecture was planned in the first place](https://www.notion.so/learningequality/Data-and-content-architecture-proposal-8e03a6da67394c578d40336b7a7953bf#af26a68dc3cb4121bf493a6c85bf6c59). We give control to the user with the storage limit setting, so if space is a concern, they can use that to manage it.

**Comment by Baizkunthaz** [comment]
Thanks a lot, @marcellamaki mam for your kind and genuine reply, but I was wondering if Python and JS would be enough to contribute to my level best to this org... or should I need to learn anything extra(I mean any other tech stack ). I would be grateful if you would suggest or guide me towards learning any required tech stack for your organization.....and I will try my best to understand the tech stack within a short duration of time...

**Comment by marcellamaki** [comment]
Hi @Baizkunthaz - you can definitely get started with contributing with Python and JS, as these are the main two languages that we use. On the front end, we use Vue, so there would be opportunities for learning there, but there are plenty of bugs that you could get started helping with as a way to do that. One way to get familiar with the types of bugs and what might be needed is to look for issues with the "help wanted" label, and you can start exploring a few to see if any seem suitable for your background. You are welcome to pick up a small issue (look for the tags "tiny" and "small"), ask if you can be assigned, and give it a try!

The [Github discussions page](https://github.com/learningequality/kolibri/discussions) I mentioned earlier is the place where you can post and ask other community contributors how they got started and if they have any tips or recommendations. If you have further general questions about our tech stack, contributing, or Kolibri, please ask them on that [discussions page](https://github.com/learningequality/kolibri/discussions) rather than on github issue threads, so we can try to keep conversations on the github issue as a more focused discussion on the issues themselves. Thank you again for your interest in contributing, and I hope we will see a contribution from you!
